File: apps/backend-site/app/schemas/post.py
from __future__ import annotations
from uuid import UUID
from pydantic import BaseModel
from datetime import datetime
from app.schemas.author import AuthorRead
from app.schemas.category import CategoryRead
from app.schemas.block import BlockCreate, BlockRead   # <–– importe BlockCreate

class PostBase(BaseModel):
    title: str
    slug: str
    author_id: UUID
    category_ids: list[UUID] = []
    thumbnail_url: str | None = None 

class PostCreate(PostBase):
    # use BlockCreate aqui: não precisa de `id`
    blocks: list[BlockCreate]

class PostUpdate(BaseModel):
    title: str | None = None
    slug: str | None = None
    author_id: UUID | None = None
    category_ids: list[UUID] | None = None
    thumbnail_url: str | None = None
    # idem para update: blocos novos sempre sem `id`
    blocks: list[BlockCreate] | None = None

class PostRead(BaseModel):
    id: UUID
    title: str
    slug: str
    author: AuthorRead
    categories: list[CategoryRead]
    blocks: list[BlockRead]
    created_at: datetime
    updated_at: datetime

    class Config:
        orm_mode = True

File: apps/portal-client/src/types/post.ts
import type { Author } from './author'
import type { Category } from './categoryBlog'

export type BlockType = 'text' | 'image' | 'video' // adicione outros se tiver

export interface Block {
  id: string
  position: number
  type: BlockType
  content: string
}

export interface BlockCreate {
  position: number
  type: BlockType
  content: string
}

export interface Post {
  id: string
  title: string
  slug: string
  author: Author
  categories: Category[]
  blocks: Block[]
  created_at: string
  updated_at: string
  thumbnail_url?: string
}

export interface PostCreate {
  title: string
  slug: string
  author_id: string
  category_ids: string[]
  blocks: BlockCreate[]
  thumbnail_url?: string
}

export interface PostUpdate {
  title?: string
  slug?: string
  author_id?: string
  category_ids?: string[]
  blocks?: BlockCreate[]
  thumbnail_url?: string
}


export interface PostPage {
  items: Post[];
  page: number;
  page_size: number;
  total: number;
}
